src/components/CursorInfo/CursorInfoComponent.tsx

```diff
@@ -139,24 +140,30 @@ export class CursorInfoComponent extends React.Component<WidgetProps> {
         const dataType = CARTA.ColumnType.String;
         const columnHeaders = columnNames.map((name, index) => new CARTA.CatalogHeader({name: name, dataType, columnIndex: index}));
 
-        const imageNames = appStore.frames.map(frame => frame.filename);
-        let values = Array(frameNum).fill("-");
-        let systems = Array(frameNum).fill("-");
-        let worldCoords = Array(frameNum).fill("-");
-        let imageCoords = Array(frameNum).fill("-");
-        const zCoords = appStore.frames.map(frame => this.genZCoordContent(frame));
-        const channels = appStore.frames.map(frame => frame.requiredChannel);
-        const stokes = appStore.frames.map(frame => frame.requiredPolarizationInfo);
+        const imageNames = appStore.imageViewConfigStore.imageNames;
+        let values = Array(imageNum).fill("-");
+        let systems = Array(imageNum).fill("-");
+        let worldCoords = Array(imageNum).fill("-");
+        let imageCoords = Array(imageNum).fill("-");
+        let zCoords = Array(imageNum).fill("-");
```

Comment:
do we need ``let`` if we are only modifying the elements of the Array?

Reply:
This is changed.